### PR TITLE
Added new test data for the CRS-2069 suppression of HDCED/4 and ERSED.

### PR DESCRIPTION
**Pull request by joelstobart-moj**
This includes JSON inputs and expected outputs, to enhance the coverage of overall calculation and SDS Early Release rules. Updated the `SDSEarlyReleaseDefaultingRulesService` to incorporate `TrancheOne` parameters in the calculation logic, ensuring correct handling of tranche dates in early release scenarios. Refactored the `handleCRDEqualsEligibilityDateAndTrancheDate` method to remove eligibility dates that coincide with controlling and tranche commencement dates. Adjusted existing test cases and added new ones to validate these changes, ensuring robustness in the handling of SDS Early Release scenarios, particularly with the introduction of the tranche dates, to allow for that. Additionally, removed unnecessary eligibility (mainly HDCED4PLUS) labels from test responses to align with updated logic.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/CalculationService.kt [hunk 1]
```diff
@@ -60,8 +69,17 @@ class CalculationService(
     trancheCommencementDate: LocalDate?,
     tranche: SDSEarlyReleaseTranche,
     trancheOneCommencementDate: LocalDate,
+    trancheDates: TrancheOne,
```
Comment (srowlands-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/CalculationService.kt:72]: calles tranche dates, but it's actually the tranche itself?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SDSEarlyReleaseDefaultingRulesService.kt [hunk 2]
```diff
@@ -98,6 +105,26 @@ class SDSEarlyReleaseDefaultingRulesService(
...
+        ReleaseDateType.ERSED,
+      )
+
+      eligibilityDates.forEach { dateType ->
+        if (dates[dateType] == controllingDate.second &&
+          (dates[dateType] == tranche.trancheCommencementDate || dates[dateType] == tranche.trancheTwoCommencementDate)
+        ) {
+          dates.remove(dateType)
+        }
+      }
```
Comment (srowlands-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SDSEarlyReleaseDefaultingRulesService.kt:124]: I'm sure Luke will be pleased that some of the breakdown changes won't ever be seen :)

**Review by srowlands-moj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/CalculationService.kt (hunk 1 above)
Comment (joelstobart-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/CalculationService.kt]: As discussed - will raise a tech debt ticket - and have renamed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SDSEarlyReleaseDefaultingRulesService.kt (hunk 2 above)
Comment (joelstobart-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SDSEarlyReleaseDefaultingRulesService.kt:124]: Shhhh

**Pull request merged**